docs/examples/ex01.py

```diff
@@ -1,3 +1,4 @@
+from pathlib import Path
```

Comment:
This breaks the line number based `literalinclude` in ex01.rst. Either fix the line numbers there or add keyword argument to `Mesh.savefig` which expands the filename using pathlib. E.g. `m.savefig("{}_solution.png", expand_stem=True)` or `m.savefig("{}/ex01_solution.png", expand_dir=True)`. If you decide to modify `Mesh.savefig` then feel free to experiment with different parameters. I'm happy as long as the changes are backwards compatible.

Reply:
> This breaks the line number based literalinclude in ex01.rst.

Oops.  Sorry.  I'll fix that in a moment.

Reply:
> which expands the filename using pathlib. E.g. m.savefig("{}_solution.png", expand_stem=True) or m.savefig("{}/ex01_solution.png", expand_dir=True). If you decide to modify Mesh.savefig then feel free to experiment with different parameters. I'm happy as long as the changes are backwards compatible.

This is a bit tricky.  It's not [`pathlib.Path`](https://docs.python.org/3/library/pathlib.html#pathlib.Path) that does the magic, it's the builtin `__file__`; I had begun typing the following into `Mesh.savefig`:
```python
        if expand_dir:
            path = path.format(Path(__file__).parent)
 ```
when I realized that there `__file__` would expand to `".../skfem/mesh/mesh.py"` which is not what's wanted.

The main problem here is to make the examples work correctly regardless of where they're called from, loading meshes
https://github.com/kinnala/scikit-fem/blob/c8ad891eb94532aa51bd634b6bd9a05622f81400/docs/examples/ex21.py#L8
 and saving figures 

 https://github.com/kinnala/scikit-fem/blob/c8ad891eb94532aa51bd634b6bd9a05622f81400/docs/examples/ex23.py#L97

from and to the directory containing the script, `…/scikit-fem/docs/examples`.  The only two ways I can think of doing this are via `sys.argv[0]` (which will contain the path of the script as a str) as in 
https://github.com/kinnala/scikit-fem/blob/c8ad891eb94532aa51bd634b6bd9a05622f81400/docs/examples/ex17.py#L87
and `__file__` as above.